src/Components/Components/src/ComponentBase.cs

```diff
@@ -102,7 +104,7 @@ protected void StateHasChanged()
                 return;
             }
 
-            if (_hasNeverRendered || ShouldRender())
+            if (_hasNeverRendered || (_hotReloadContext?.IsHotReloading() ?? false) || ShouldRender())
```

Comment:
If there's some way to tell the linker to treat `_hotReloadContext` as always-null that would be pretty neat!

Or is it somehow smart enough to do that based on the absence of any calls into `IReceiveHotReloadContext.Receive`? That would be super impressive if so.